### PR TITLE
Simplify fluid api recognize digit

### PR DESCRIPTION
**Pull request by jetfuel**

### CODE REVIEW

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+import unittest
+import math
+import sys
+import os
+import paddle.v2.dataset as dataset
+
+BATCH_SIZE = 64
+
+
+def inference_network():
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: Could you rename `inference_network` to `inference_program` and `train_network` to `train_program`? Yi pointed out that "program" is more suitable for Fluid than "network".

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
...
+            if (event.batch_id + 1) % 10 == 0:
+                test_metrics = trainer.test(reader=dataset.mnist.test())
+                avg_loss_set = test_metrics[0]
+                acc_set = test_metrics[1]
+
+                # get test acc and loss
+                acc_val = numpy.array(acc_set).mean()
+                avg_loss_val = numpy.array(avg_loss_set).mean()
+                if float(acc_val) > 0.2:  # Smaller value to increase CI speed
+                    trainer.params.save(save_dirname)
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: Can you change `trainer.params.save` to `trainer.save_params`? (https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/book/word2vec/no_test_word2vec_new_api.py#L103)

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+
+def train(use_cuda, save_dirname):
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+
+    optimizer = fluid.optimizer.Adam(learning_rate=0.001)
+    trainer = fluid.Trainer(train_network, place=place, optimizer=optimizer)
+
+    def event_handler(event):
+        if isinstance(event, fluid.EndIteration):
+            print(event.metrics)
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: Can you change `print(event.metrics)` to `avg_cost, acc = event.values`? (e.g., same name as the return value of `train_network`), and rename `avg_loss_val` and `acc_val` below to `avg_cost` and `acc` for clarity.

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+
+    trainer.train(
+        reader=dataset.mnist.train(), num_pass=100, event_handler=event_handler)
+
+
+def infer(use_cuda, save_dirname=None):
+    params = fluid.Params(save_dirname)
+    place = fluid.CUDAPlace(0) if use_cuda else fluid.CPUPlace()
+
+    inferencer = fluid.Inferencer(inference_network, params, place=place)
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: Sorry that the API has changed slightly, please change `Inferencer` and `Trainer`'s constructor according to https://github.com/PaddlePaddle/Paddle/blob/develop/python/paddle/fluid/tests/book/word2vec/no_test_word2vec_new_api.py#L118

Comment (jetfuel) [reply python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: No Problem, will do

**Comment by jetfuel** [comment]
@helinwang Updated according to review.

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py [hunk 5]
```diff
@@ -0,0 +1,120 @@
...
+                test_metrics = trainer.test(reader=dataset.mnist.test())
+                avg_cost_set = test_metrics[0]
+                acc_set = test_metrics[1]
+
+                # get test acc and loss
+                acc = numpy.array(acc_set).mean()
+                avg_cost = numpy.array(avg_cost_set).mean()
+                if float(acc) > 0.2:  # Smaller value to increase CI speed
+                    trainer.save_params(save_dirname)
+                    return
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: This `return` actually does not stop training, perhaps remove it.
Not related to this PR, maybe we need a function called `trainer.stop_train`.

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py [hunk 6]
```diff
@@ -0,0 +1,120 @@
...
+                else:
+                    print(
+                        'BatchID {1:04}, Test Loss {2:2.2}, Acc {3:2.2}'.format(
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: are `1:04`, `2:2.2`, `3:2.2` typos?

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_mlp.py [hunk 7]
```diff
@@ -0,0 +1,108 @@
...
+                # get test acc and loss
+                acc = numpy.array(acc_set).mean()
+                avg_cost = numpy.array(avg_cost_set).mean()
+                if float(acc) > 0.2:  # Smaller value to increase CI speed
+                    trainer.save_params(save_dirname)
+                    return
+
+                else:
+                    print(
+                        'BatchID {1:04}, Test Loss {2:2.2}, Acc {3:2.2}'.format(
```
Comment (helinwang) [inline python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_mlp.py]: are `1:04`, `2:2.2`, `3:2.2` typos?

File: python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py (hunk 6 above)
Comment (jetfuel) [reply python/paddle/fluid/tests/book/notest_recognize_digits/notest_recognize_digits_conv.py]: Good catch! They are from the original code. When I removed the first item in the print, I forgot to change the others. Will fix this.

**Pull request merged**